relay/server.go

```diff
@@ -0,0 +1,196 @@
+package relay
+
+import (
+	"context"
+	"fmt"
+	pb "github.com/Layr-Labs/eigenda/api/grpc/relay"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/disperser/common/v2/blobstore"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/relay/chunkstore"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+var _ pb.RelayServer = &Server{}
+
+// Server implements the Relay service defined in api/proto/relay/relay.proto
+type Server struct {
+	pb.UnimplementedRelayServer
+
+	// metadataServer encapsulates logic for fetching metadata for blobs.
+	metadataServer *metadataManager
+
+	// blobServer encapsulates logic for fetching blobs.
+	blobServer *blobManager
+
+	// chunkServer encapsulates logic for fetching chunks.
+	chunkServer *chunkManager
+}
+
+// NewServer creates a new relay Server.
+func NewServer(
+	ctx context.Context,
+	logger logging.Logger,
+	config *Config,
+	metadataStore *blobstore.BlobMetadataStore,
+	blobStore *blobstore.BlobStore,
+	chunkReader chunkstore.ChunkReader) (*Server, error) {
+
+	ms, err := newMetadataManager(
+		ctx,
+		logger,
+		metadataStore,
+		config.MetadataCacheSize,
+		config.MetadataWorkPoolSize,
+		config.Shards)
+	if err != nil {
+		return nil, fmt.Errorf("error creating metadata server: %w", err)
+	}
+
+	bs, err := newBlobManager(
+		ctx,
+		logger,
+		blobStore,
+		config.BlobCacheSize,
+		config.BlobWorkPoolSize)
+	if err != nil {
+		return nil, fmt.Errorf("error creating blob server: %w", err)
+	}
+
+	cs, err := newChunkManager(
+		ctx,
+		logger,
+		chunkReader,
+		config.ChunkCacheSize,
+		config.ChunkWorkPoolSize)
+	if err != nil {
+		return nil, fmt.Errorf("error creating chunk server: %w", err)
+	}
+
+	return &Server{
+		metadataServer: ms,
+		blobServer:     bs,
+		chunkServer:    cs,
+	}, nil
+}
+
+// GetBlob retrieves a blob stored by the relay.
+func (s *Server) GetBlob(ctx context.Context, request *pb.GetBlobRequest) (*pb.GetBlobReply, error) {
+
+	// Future work	:
+	//  - global throttle
+	//  - per-connection throttle
+	//  - timeouts
+
+	keys := []v2.BlobKey{v2.BlobKey(request.BlobKey)}
```

Comment:
using `corev2.BytesToBlobKey` will also validate the key

Reply:
Neat, I've added this to all places where an end user can supply a blob key